modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/Metadata.java

```diff
@@ -76,16 +76,16 @@ public Metadata(String routingKey, String exchangeName, long contentLength) {
         this.routingKey = routingKey;
         this.exchangeName = exchangeName;
         this.contentLength = contentLength;
-        this.properties = FieldTable.EMPTY_TABLE;
-        this.headers = FieldTable.EMPTY_TABLE;
+        this.properties = new FieldTable();
+        this.headers = new FieldTable();
     }
 
     public Metadata(String routingKey, String exchangeName, long contentLength, byte[] propertyBytes) throws Exception {
         this.routingKey = routingKey;
         this.exchangeName = exchangeName;
         this.contentLength = contentLength;
-        this.properties = FieldTable.EMPTY_TABLE;
-        this.headers = FieldTable.EMPTY_TABLE;
+        this.properties = new FieldTable();
+        this.headers = new FieldTable();
```

Comment:
Can we call `this(routingKey, exchangeName, contentLength)` here? It will help to eliminate duplicate code.